File: insight-brain-frontend/src/main/frontend/owner.manager/innersource.repository/innersource.repository.tile.js
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
import { actions as innerSourceRepositoryBaseConfigurationsActions } from 'MainRoot/innerSourceRepositoryConfiguration/innerSourceRepositoryBaseConfigurationsSlice';
import { selectIsInnerSourceRepositorySupported } from 'MainRoot/productFeatures/productFeaturesSelectors';
import template from './innersource.repository.tile.html';
import { selectIsApplication, selectIsOrganization } from 'MainRoot/reduxUiRouter/routerSelectors';
import { selectSelectedOwnerId } from 'MainRoot/OrgsAndPolicies/orgsAndPoliciesSelectors';
import {
  selectEditLink,
  selectInheritedFromOrganizationName,
  selectInnerSourceRepositoriesEnabled,
  selectLoadError,
  selectLoading,
  selectRepositoryConnections,
} from 'MainRoot/innerSourceRepositoryConfiguration/innerSourceRepositoryBaseConfigurationsSelectors';
export default {
  template: template,
  controllerAs: 'vm',
  controller: InnerSourceRepositoryTileController,
};

function InnerSourceRepositoryTileController($scope, $ngRedux) {
  var vm = this;
  vm.doLoad = doLoad;

  vm.unsubscribe = $ngRedux.connect(mapStateToThis, {
    loadRepositoryConnections: innerSourceRepositoryBaseConfigurationsActions.load,
  })(vm);

  $scope.$on('$destroy', function () {
    vm.unsubscribe();
  });

  vm.doLoad();

  function doLoad() {
    if (vm.isInnerSourceRepositorySupported) {
      vm.loadRepositoryConnections({ ownerId: vm.ownerId, inherit: true });
    }
  }
}

const mapStateToThis = (state) => ({
  isInnerSourceRepositorySupported: selectIsInnerSourceRepositorySupported(state),
  isOrg: selectIsOrganization(state),
  isApp: selectIsApplication(state),
  ownerId: selectSelectedOwnerId(state),
  editLink: selectEditLink(state),
  loading: selectLoading(state),
  error: selectLoadError(state),
  innerSourceRepositories: angular.copy(selectRepositoryConnections(state)),
  innerSourceRepositoriesInheritedFrom: selectInheritedFromOrganizationName(state),
  innerSourceRepositoriesEnabled: selectInnerSourceRepositoriesEnabled(state),
});

InnerSourceRepositoryTileController.$inject = ['$scope', '$ngRedux'];

File: insight-brain-testing/insight-brain-functional-test-common/src/main/java/com/sonatype/clm/testing/functional/elements/InnerSourceRepositoryTile.java
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
package com.sonatype.clm.testing.functional.elements;

import com.codeborne.selenide.ElementsCollection;
import com.codeborne.selenide.SelenideElement;

public class InnerSourceRepositoryTile
    extends OwnerTile
{
  private static final String INNERSOURCE_REPOSITORY_OWNER_ELEMENT_ID = "#owner-pill-innersource-repository";

  public InnerSourceRepositoryTile() {
    super(INNERSOURCE_REPOSITORY_OWNER_ELEMENT_ID);
  }

  public ElementsCollection rows() {
    return children(".nx-list__item");
  }

  public SelenideElement listTitle() {
    return child(".nx-h3");
  }

  public SelenideElement editButton() {
    return child("#innersource-repositories-edit");
  }
}
